tests/ethercat/test_pdo.py

```diff
@@ -281,18 +289,28 @@ def test_start_stop_pdo(connect_to_all_slave):
             register = servo.dictionary.registers(SUBNODE)[rpdo_register]
             rpdo_map.add_registers(register)
         for item in rpdo_map.items:
-            item.value = 0
+            item.value = initial_positions[index]
         servo.set_pdo_map_to_slave([rpdo_map], [tpdo_map])
         net._ecat_master.read_state()
         assert servo.slave.state_check(pysoem.PREOP_STATE) == pysoem.PREOP_STATE
     net.start_pdos()
     net._ecat_master.read_state()
     for servo in servos:
         assert servo.slave.state_check(pysoem.OP_STATE) == pysoem.OP_STATE
+    start_time = time.time()
+    timeout = 2
+    while time.time() < start_time + timeout:
+        for servo in servos:
+            for item in servo._rpdo_maps[0].items:
+                item.value += 100
```

Comment:
Move the motor is not necessary, the only check that must be done is to ensure the PDO value is right, you can compare the PDO value with an SDO value. Moving the motor can be problematic.

Reply:
Done